src/Core/FeatureFlag/FeatureFlagsModifier.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\FeatureFlag;
+
+use Doctrine\ORM\EntityManagerInterface;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShopBundle\Entity\FeatureFlag;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * This class is in charge of controlling the state modifications of Feature Flags
+ * through the Back Office interface. It fetches the data needed for the Back Office interface
+ * and handles payloads from the Back Office page, replicating the changes on the data model.
+ */
+class FeatureFlagsModifier implements DataConfigurationInterface
+{
+    /** @var EntityManagerInterface */
+    private $doctrineEntityManager;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /**
+     * @param EntityManagerInterface $doctrineEntityManager
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(EntityManagerInterface $doctrineEntityManager, TranslatorInterface $translator)
+    {
+        $this->doctrineEntityManager = $doctrineEntityManager;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration(): array
+    {
+        $configuration = [];
+
+        foreach ($this->getAllFeatureFlags() as $id => $featureFlag) {
+            $configuration[$featureFlag->getName()] = $featureFlag->getState();
```

Comment:
```suggestion
            $configuration[$featureFlag->getId()] = $featureFlag->getState();
```
Shouldn't be better to identify the component with its id?

Reply:
The name of a feature flag is a slug like `product_page_v2` so it sounds like a better identifier than a SQL ID 🤔  ?

Both can be used as an identifier.

Reply:
Ohh, I usually use `name` for something that can change in time :sweat_smile: 
I've the habit of use `slug`/`identifier`.
Otherwise, you have to put a Uniq constraint here because if someone add a new feature flag, you have to be sure it's not a duplicate, or it will break what you define here
https://github.com/PrestaShop/PrestaShop/blob/e753b57bfb9839d5d9e034cf71414591f721f445/src/PrestaShopBundle/Entity/FeatureFlag.php#L56

Reply:
The name is unique, so it's ok :)

Reply:
> I've the habit of use slug/identifier.

I agree with you. I ask @eternoendless as he decided the SQL structure
@eternoendless would you be OK to rename the `name` column in `identifier`?

Reply:
I don't have a preference. Just avoid `id` because it's usually an integer.